### PR TITLE
Implemented hook for inspector component

### PR DESCRIPTION
**Pull request by sashkivna**
Implemented login for select element from rendered in canvas component

### CODE REVIEW

File: scripts/main/designer/components/inspector/inspector.directive.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        return {
+            restrict: 'E',
+            templateUrl: 'main/designer/components/inspector/inspector.template.html',
+
+            scope: {
+                component: '=',
+            },
+
+            link: function ($scope) {
+                $scope.$watch('component', function (component) {
```
Comment (r0ss0neri) [inline scripts/main/designer/components/inspector/inspector.directive.js]: eliminate `watch`

File: scripts/main/designer/components/stencil/stencil.directive.js [hunk 2]
```diff
@@ -13,8 +13,6 @@ angular.module('designer.module')
                 $scope.selectedComponent = null;
 
                 $scope.addSelectedComponent = function () {
-                    //console.log($scope.selectedComponent);
```
Comment (r0ss0neri) [inline scripts/main/designer/components/stencil/stencil.directive.js]: eliminate

File: scripts/main/designer/components/stencil/stencil.template.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
-<div ng-repeat="component in components">
+<div ng-repeat="component in components track by $index">
```
Comment (r0ss0neri) [inline scripts/main/designer/components/stencil/stencil.template.html]: don't need `track by $index"`

File: scripts/main/designer/designer.directive.js [hunk 4]
```diff
@@ -6,21 +6,36 @@ angular.module('designer.module')
...
+
                 $scope.inspectorComponent = null;
+                $scope.displayInspector = false;
+
+                $scope.definitions = {
+                    comments: []
+                };
+
+                $scope.addComponentToCanvas = function (componentDescriptor) {
+                    debugger;
```
Comment (r0ss0neri) [inline scripts/main/designer/designer.directive.js]: remove debugger

File: scripts/main/designer/components/inspector/inspector.template.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div class="inspector-look">
```
Comment (r0ss0neri) [inline scripts/main/designer/components/inspector/inspector.template.html]: `div` can be eliminated

File: scripts/styles/styles.css [hunk 6]
```diff
@@ -39,9 +39,13 @@ my-stencil .selected-component {
...
-    width: 100%;
+    height: 50px;
+    width: 95%;
     background-color: #F6C6CE;
     border: 1px solid rebeccapurple;
-    margin: 2px;
+    margin: 2px 2px 2px 15px;
+}
+
+.inspector-look {
```
Comment (r0ss0neri) [inline scripts/styles/styles.css]: inspector {
    ...
    }

File: scripts/main/designer/components/canvas/canvas.directive.js [hunk 7]
```diff
@@ -12,9 +12,11 @@ angular.module('designer.module')
             link: function ($scope) {
                 $scope.selectedComponent = null;
 
-                $scope.selectFromCanvas = function (component) {
+                $scope.selectFromCanvas = function (component, $index) {
                     $scope.selectedComponent = component;
-                    $scope.onSelect({component: $scope.selectedComponent});
+                    $scope.onSelect({component: $scope.selectedComponent,
+                                     $index: $index
```
Comment (r0ss0neri) [inline scripts/main/designer/components/canvas/canvas.directive.js]: looks like we don't need `$index`

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: scripts/main/designer/components/inspector/inspector.directive.js (hunk 1 above)
Comment (sashkivna) [reply scripts/main/designer/components/inspector/inspector.directive.js]: done

File: scripts/main/designer/components/stencil/stencil.template.html (hunk 3 above)
Comment (sashkivna) [reply scripts/main/designer/components/stencil/stencil.template.html]: removed

File: scripts/main/designer/designer.directive.js (hunk 4 above)
Comment (sashkivna) [reply scripts/main/designer/designer.directive.js]: done

File: scripts/styles/styles.css (hunk 6 above)
Comment (sashkivna) [reply scripts/styles/styles.css]: done

File: scripts/main/designer/components/canvas/canvas.directive.js (hunk 7 above)
Comment (sashkivna) [reply scripts/main/designer/components/canvas/canvas.directive.js]: we are tracking all canvas elements by index and after clicking we can use appropriate model

File: scripts/main/designer/components/inspector/inspector-component/inspector-component.directive.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+                propertyConfig: '='
+            },
+
+            link: function ($scope, $element) {
+                $element.append($compile('<' + $scope.propertyConfig.type +"-field"+ '></' + $scope.propertyConfig.type + "-field"+'>')($scope));
+            }
+        };
+    });
+
+//<input ng-model="model[propertyConfig.name]" type="' + $scope.propertyConfig.type + '"/>
```
Comment (r0ss0neri) [inline scripts/main/designer/components/inspector/inspector-component/inspector-component.directive.js]: remove

File: scripts/main/designer/components/stencil/stencil.template.html [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <div ng-repeat="component in components">
     <div class="component"
          ng-class="{'selected-component': isComponentSelected(component)}">
-        <label ng-click="selectComponent(component)">{{component.label}}</label>
+        <label ng-click="selectComponent(component, $index)">{{::component.componentName}}</label>
```
Comment (r0ss0neri) [inline scripts/main/designer/components/stencil/stencil.template.html]: remove `$index`

File: scripts/main/designer/designer.directive.js [hunk 10]
```diff
@@ -1,23 +1,46 @@
...
-                $scope.stencilComponents = [fooComponentDescription];
                 $scope.canvasComponents = [];
 
-                $scope.addComponentToCanvas = function (component) {
-                    $scope.canvasComponents.push(component);
+                $scope.inspectorComponent = null;
+                $scope.displayInspector = false;
+
+                $scope.definitions = {
+                    comments: []
```
Comment (r0ss0neri) [inline scripts/main/designer/designer.directive.js]: rename to `components`

**Review by r0ss0neri — CHANGES_REQUESTED** [review]

File: scripts/main/designer/components/inspector/inspector-component/inspector-component.directive.js (hunk 8 above)
Comment (sashkivna) [reply scripts/main/designer/components/inspector/inspector-component/inspector-component.directive.js]: done

File: scripts/main/designer/components/stencil/stencil.template.html (hunk 9 above)
Comment (sashkivna) [reply scripts/main/designer/components/stencil/stencil.template.html]: done

**Review by r0ss0neri — APPROVED** [review]

**Pull request merged**